app/queries/complete_orders_with_balance.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Fetches complete orders of the specified user including their balance as a computed column
+class CompleteOrdersWithBalance
+  def initialize(user)
+    @user = user
+  end
+
+  def query
+    OutstandingBalance.new(sorted_complete_orders).query
+  end
+
+  private
+
+  def sorted_complete_orders
+    @user.orders
+      .where.not(Spree::Order.in_incomplete_state.where_values_hash)
```

Comment:
:thinking:

Reply:
If we go with the `finalized` / `unfinalized` scopes, this whole line can can just be `.finalized`.

Reply:
I'll try removing the double negation once I addressed the scope naming but [where_values_hash](https://apidock.com/rails/ActiveRecord/Relation/where_values_hash) is the way to get the WHERE from the underlying AR Relation object to pass them to `not`. Apparently, there's no way to make `not` and `merge` work together.